lib/package/Endpoint.js

```diff
@@ -137,117 +138,110 @@ Endpoint.prototype.getProxyName = function() {
   return this.proxyName;
 };
 
-Endpoint.prototype.getPreFlow = function() {
-  if (!this.preFlow) {
-    var doc = xpath.select("./PreFlow", this.element);
-    if (doc && doc[0]) {
-      this.preFlow = new Flow(doc[0], this);
-    }
+Endpoint.prototype.getPreFlow = function () {
+  debug("> getPreFlow()");
+  if (this.preFlow == null) {
+    const n = xpath.select("./PreFlow", this.element);
+    this.preFlow = n && n[0] && new Flow(n[0], this);
   }
+  debug("< getPreFlow()");
   return this.preFlow;
 };
 
-Endpoint.prototype.getPostFlow = function() {
-  if (!this.postFlow) {
-    var doc = xpath.select("./PostFlow", this.element);
-    if (doc && doc[0]) {
-      this.postFlow = new Flow(doc[0], this);
-    }
+Endpoint.prototype.getPostFlow = function () {
+  debug("> getPostFlow()");
+  if (this.postFlow == null) {
+    const n = xpath.select("./PostFlow", this.element);
+    this.postFlow = n && n[0] && new Flow(n[0], this);
   }
+  debug("< getPostFlow()");
   return this.postFlow;
 };
 
-Endpoint.prototype.getPostClientFlow = function() {
-  if (!this.postClientFlow) {
-    var doc = xpath.select("./PostClientFlow", this.element);
-    if (doc && doc[0]) {
-      this.postClientFlow = new Flow(doc[0], this);
-    }
+Endpoint.prototype.getPostClientFlow = function () {
+  debug("> getPostClientFlow()");
+  if (this.postClientFlow == null) {
+    const n = xpath.select("./PostClientFlow", this.element);
+    this.postClientFlow = n && n[0] && new Flow(n[0], this);
   }
+  debug("< getPostClientFlow()");
   return this.postClientFlow;
 };
 
-Endpoint.prototype.getSharedFlow = function() {
-  if (!this.sharedFlow) {
-    var doc = xpath.select(".", this.element);
-    if (doc && doc[0]) {
-      this.sharedFlow = new Flow(doc[0], this);
-    }
+Endpoint.prototype.getSharedFlow = function () {
+  debug("> getSharedFlow()");
+  if (this.sharedFlow == null) {
+    const n = xpath.select("/SharedFlow", this.element);
```

Comment:
shouldnt it be `"./SharedFlow"`

Reply:
It should not.  

In the "normal" Endpoint cases the `Flow` element holds `Request` and/or `Response`, which then hold `Step` elements.  For each of `getFlowRequest()`, `getFlowResponse()`, and `getSharedFlow()`, we want to return the thing that holds `Step` elements. Therefore in the case of `.getFlowRequest()`, the selector is `./Request`, and for `.getFlowResponse()`, the selector is `./Response`.        apigeelint re-uses (overloads?) the Endpoint type to also hold SharedFlows, and in that case, `Step` elements are children of the toplevel element.  So we want `/SharedFlow` there.  This retrieves the toplevel element, if and only if it is `SharedFlow`.